File: src/main/java/cqrs/mr/commandhandlers/DeactivateInventoryItemHandler.java
package cqrs.mr.commandhandlers;

import cqrs.core.Handler;
import cqrs.core.Repository;
import cqrs.mr.commands.DeactivateInventoryItem;
import cqrs.mr.domain.InventoryItem;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class DeactivateInventoryItemHandler implements Handler<DeactivateInventoryItem> {
	private final Repository<InventoryItem> repository;

	@Autowired
	public DeactivateInventoryItemHandler(Repository<InventoryItem> repository) {
		this.repository = repository;
	}

	@Override
	public void handle(DeactivateInventoryItem command) {
		InventoryItem item = repository.getById(InventoryItem.class,command.id);
		item.deactivate();
		repository.save(item, command.originalVersion);
	}
}

File: app/cqrs/mr/commandhandlers/DeactivateInventoryItemHandler.scala
package cqrs.mr.commandhandlers

import cqrs.core.Handler
import cqrs.core.Repository
import cqrs.mr.commands.DeactivateInventoryItem
import cqrs.mr.domain.InventoryItem

class DeactivateInventoryItemHandler(repository: Repository[InventoryItem]) extends Handler[DeactivateInventoryItem] {

  def handle(command: DeactivateInventoryItem): Unit = {
    val item = repository.getById(classOf[InventoryItem], command.id)
    item.deactivate()
    repository.save(item, command.originalVersion)
  }
}
